core/geometry/src/topology/Voronoi.ts

```diff
@@ -6,704 +6,736 @@
-  public p0: XAndY;
-  public p1: XAndY;
-  public p2: XAndY;
-  public p3: XAndY;
+  public bbox: Range2d;
   /**
-   * Constructor
-   * @param graphRange Range of the HalfEdgeGraph representing a Delaunay triangulated graph; xy-only (z-coordinate is ignored).
-   * @param circumcenterRange Range of the circumcenters of the triangles in the Delaunay triangulation.
+   * Constructor that takes up to two ranges to union and expand by a margin.
+   * * For example, Delaunay graph range and circumcenter range.
    */
-  constructor(graphRange: Range3d, circumcenterRange: Range3d) {
-    const minPadding = 2;  // smallest padding
-    const maxPadding = 10; // do not let padding grow too large
-    const ratio = 0.05;    // 5% relative padding
-    const padX = Math.min(Math.max(graphRange.xLength() * ratio, minPadding), maxPadding) + circumcenterRange.xLength();
-    const padY = Math.min(Math.max(graphRange.yLength() * ratio, minPadding), maxPadding) + circumcenterRange.yLength();
-    this.p0 = Point2d.create(graphRange.low.x - padX, graphRange.low.y - padY);
-    this.p1 = Point2d.create(graphRange.high.x + padX, graphRange.low.y - padY);
-    this.p2 = Point2d.create(graphRange.high.x + padX, graphRange.high.y + padY);
-    this.p3 = Point2d.create(graphRange.low.x - padX, graphRange.high.y + padY);
-  }
-  public contains(point: Point3d): boolean {
-    return this.p0.x <= point.x && point.x <= this.p1.x &&
-      this.p1.y <= point.y && point.y <= this.p2.y;
-  }
-  public intersect(line: Line): Point3d[] {
-    const intersections: Point3d[] = [];
-    const fractions: Vector2d = Vector2d.createZero();
-    for (const pair of [[this.p0, this.p1], [this.p1, this.p2], [this.p2, this.p3], [this.p3, this.p0]]) {
-      const intersectionFound = SmallSystem.lineSegment2dXYTransverseIntersectionUnbounded(
-        line.start as Point2d, line.end as Point2d, pair[0] as Point2d, pair[1] as Point2d, fractions,
-      );
-      if (!intersectionFound)
-        continue; // no intersection found
-      const lineFraction = fractions.x;
-      const pairFraction = fractions.y;
-      if (lineFraction >= 0 && lineFraction <= 1 && pairFraction >= 0 && pairFraction <= 1) {
-        const direction = Vector2d.createStartEnd(line.start, line.end);
-        const d: XYAndZ = { x: direction.x, y: direction.y, z: 0 };
-        intersections.push(Point3d.createAdd2Scaled(line.start as XYAndZ, 1, d, lineFraction));
-      }
+  public constructor(r0: LowAndHighXY, r1?: LowAndHighXY) {
+    this.bbox = Range2d.createFrom(r0);
+    if (r1)
+      this.bbox.union(r1, this.bbox);
+    if (!this.bbox.isNull && !this.bbox.isSinglePoint) {
+      const pad = 0.02473 * (this.bbox.xLength() + this.bbox.yLength()); // ~5% of average side length
+      this.bbox.expandInPlace(pad);
     }
-    return intersections;
+  }
+  /** Checks if a point is contained within the boundary. */
+  public contains(point: XAndY): boolean {
+    return this.contains(point);
```

Comment:
how is this possible? how it's not creating an infinite recursion?!

Reply:
Ha! Indeed it would be infinite, if it were ever called! I'll remove.